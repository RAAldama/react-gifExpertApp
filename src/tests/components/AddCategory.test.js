import { shallow } from "enzyme";
import AddCategory from '../../components/AddCategory';

describe('Test on <Add Category />', () => {
    
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    });

    test('should show correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should change the input text box', () => {
        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        
        input.simulate('change', { target: { value } });
        
        const para = wrapper.find('p').text().trim();
        expect(para).toBe(value);
    })

    test('should not post the info onSubmit', () => {
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect(setCategories).not.toHaveBeenCalled();
    })

    test('should call the setCategories and clear the text box', () => {
        const value = 'OwO';
        const input = wrapper.find('input');
        input.simulate('change', { target: { value } });

        const form = wrapper.find('form');
        form.simulate('submit', { preventDefault(){} });

        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledWith( expect.any(Function) );
        expect( wrapper.find('input').prop('value') ).toBe('');
    })

})
