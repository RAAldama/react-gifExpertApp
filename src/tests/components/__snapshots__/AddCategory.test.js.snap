// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <Add Category /> should show correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="Ingrese CategorÃ­a"
    type="text"
    value=""
  />
</form>
`;
